### PR TITLE
Asset-Upload fixed

### PR DESCRIPTION
**Pull request by wsmirnow**
This patch fixes the asset upload.

### CODE REVIEW

File: src/i18n/org/opencastproject/adminui/languages/lang-en_US.json [hunk 1]
```diff
@@ -171,6 +171,7 @@
 		"EVENTS_CREATED": "The event has been created",
 		"EVENTS_UPLOAD_STARTED": "The event is being uploaded… {{ progress }}%",
 		"EVENTS_NOT_CREATED": "The event could not be created",
+		"EVENTS_UPDATED": "The event have been successfully updated",
```
Comment (Arnei) [inline src/i18n/org/opencastproject/adminui/languages/lang-en_US.json:174]:
```suggestion
		"EVENTS_UPDATED": "The event has been successfully updated",
```

File: src/slices/eventDetailsSlice.ts [hunk 2]
```diff
@@ -1564,24 +1564,47 @@ export const updateAssets = createAsyncThunk('eventDetails/updateAssets', async
...
+		"downloadSourceflavorsExist": string,
+		"download-source-flavors": string,
+	} = {
+		"downloadSourceflavorsExist": String(assetFlavors.length > 0),
+		"download-source-flavors": assetFlavors,
+	};
+
+	let processing: {
+		workflow: string | undefined,
+        configuration: typeof uploadAssetWorkflowConfiguration,
```
Comment (Arnei) [inline src/slices/eventDetailsSlice.ts:1597]: Should be using tabs instead of spaces for indentation.

File: src/slices/eventDetailsSlice.ts [hunk 3]
```diff
@@ -1564,24 +1564,47 @@ export const updateAssets = createAsyncThunk('eventDetails/updateAssets', async
...
+			let uploadAssetFlavor = [option.flavorType, option.flavorSubType].join("/");
+			if (assetFlavors.length > 0) {
+				assetFlavors = [assetFlavors, uploadAssetFlavor].join(",");
+			} else {
+				assetFlavors = uploadAssetFlavor;
+			}
 		}
 	});
 
+	let uploadAssetWorkflowConfiguration: {
```
Comment (Arnei) [inline src/slices/eventDetailsSlice.ts:1587]: Minor stylistic nitpick: It is preferred to use `const` instead of `let` whenever possible.

**Review by Arnei — CHANGES_REQUESTED** [review]
Works and makes sense to me.

Ideally we would not have to make assumptions about magic variables in workflows and have the backend sort it out somehow, but arguably restoring the upload to working order is more important and the assumptions we have to make are not too annoying.

Minor nitpicks below:

File: src/slices/eventDetailsSlice.ts [hunk 4]
```diff
@@ -1564,24 +1564,47 @@ export const updateAssets = createAsyncThunk('eventDetails/updateAssets', async
...
 
 	let assets: {
-		workflow: string | undefined,
 		options: UploadAssetOption[],
 	} = {
-		workflow: uploadAssetWorkflow,
 		options: [],
 	};
 
+	let assetFlavors: string = "";
```
Comment (Arnei) [inline src/slices/eventDetailsSlice.ts:1572]: It is not necessary to type everything explicitly. For example, in this case typescripts type inference would have no problem inferring that this should be a string.

File: src/slices/eventDetailsSlice.ts (hunk 2 above)
Comment (wsmirnow) [reply src/slices/eventDetailsSlice.ts]: done

File: src/slices/eventDetailsSlice.ts (hunk 3 above)
Comment (wsmirnow) [reply src/slices/eventDetailsSlice.ts]: Thank you for the info. I've updated the PR.

**Comment by wsmirnow** [comment]
Thank you @Arnei. I've updated the PR with your suggestions. Please continue review.

**Review by Arnei — APPROVED** [review]

**Pull request merged**